pkg/operation/botanist/botanist.go

```diff
@@ -150,6 +150,16 @@ func New(ctx context.Context, o *operation.Operation) (*Botanist, error) {
 	if err != nil {
 		return nil, err
 	}
+	//TODO(acumino): Need to add when node local dns is enabled
+	if o.Shoot.NodeLocalDNSEnabled {
+		o.Shoot.Components.SystemComponents.NodeLocalDNS, err = b.DefaultNodeLocalDNS()
+		if err != nil {
+			return nil, err
+		} else {
+			o.Shoot.Components.SystemComponents.NodeLocalDNS = nil
+		}
```

Comment:
You can wrap ` b.DefaultNodeLocalDNS()` with `component.OpDestroy` if the node local dns is not enabled. Another option is to manually call Destroy of the NodeLocalDNS component in shoot_control_{reconcile,delete}.